components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java

```diff
@@ -492,8 +491,13 @@ public void newNonEmptyResource(String path, boolean isDirectory, String content
             String parent = getParentPath(targetPath, isDirectory);
             String fileName = getResourceName(targetPath);
 
+            Properties metadata = new Properties();
+            if (mediaType != null) {
```

Comment:
Do we need this null check?

Reply:
Yes. This is called from FileRegistryResourceDeployer class which is in another jar and before this commit its behavior was to sent null. Its worth having a null check here.